### PR TITLE
Translates src/user/online.js to src/user/online.ts

### LINKED ISSUE #1: Translate online.js to online.ts
Issue 1A

- Estimated Effort: 3 days
- Dependencies: None
- Initial Assignments: [Ellen Fang](mailto:ellenf@andrew.cmu.edu) 
- Milestones: Completed in Sprint 1
- Acceptance Criteria:
  - Translates JS file to TS completely
  - Passes lint
  - Passes test
  - Passes workflow overall

### PR DESCRIPTION
**Pull request by ellenfang77**
Translates src/user/online.js to src/user/online.ts:

- Create interfaces for data types to avoid 'any' data types
- Uses disable Eslint comments for accesses to other files' functions that throw lint errors
       - lines 25, 39, 44, 46, 55, 58
-  Change import statements

Addresses #1

### CODE REVIEW

File: src/user/online.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        const userData: userDataTemplate = await db.getObjectFields(`user:${uid}`, ['status', 'lastonline']) as userDataTemplate;
+        const now = Date.now();
+        if (userData.status === 'offline' || now - parseInt(userData.lastonline, 10) < 300000) {
+            return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        await User.setUserField(uid, 'lastonline', now);
```
Comment (cwesterink) [inline src/user/online.ts:30]: I think this comment can be removed now

File: src/user/online.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+interface userDataTemplate {
+    status: string;
+    lastonline: string;
+}
+
+interface UserTemplate {
+    uid: string;
+    updateLastOnlineTime: (uid: string) => Promise<void>;
+    setUserField: (uid: string, field: string, value: any) => Promise<void>;
```
Comment (cwesterink) [inline src/user/online.ts]: maybe we can set the value to type `number | string`. Not a big deal

**Review by cwesterink — APPROVED** [review]
LGTM

File: src/user/online.ts (hunk 2 above)
Comment (ellenfang77) [reply src/user/online.ts]: fixed

File: src/user/online.ts (hunk 1 above)
Comment (ellenfang77) [reply src/user/online.ts:30]: fixed

**Pull request merged**